### PR TITLE
Add CERTIFIED column to package list

### PR DESCRIPTION
**Pull request by janisz**
To distinguish community and regular packages, add `CERTIFIED` column to `dcos package list` command and change `SELECTED` to `CERTIFIED` in `dcos package search`.

Allow user to filter by part of the name passed as the first argument.

Fixes:
* https://jira.mesosphere.com/browse/DCOS-55773
* https://jira.mesosphere.com/browse/DCOS_OSS-5607

### CODE REVIEW

File: pkg/cmd/pkg/package_list.go [hunk 1]
```diff
@@ -133,7 +134,7 @@ func newCmdPackageList(ctx api.Context) *cobra.Command {
 				return errors.New("there are currently no installed packages")
 			}
 
-			table := cli.NewTable(ctx.Out(), []string{"NAME", "VERSION", "APP", "COMMAND", "DESCRIPTION"})
+			table := cli.NewTable(ctx.Out(), []string{"NAME", "VERSION", "SELECTED", "APP", "COMMAND", "DESCRIPTION"})
```
Comment (makkes) [inline pkg/cmd/pkg/package_list.go]: What about taking the opportunity and add a unit test for this cmd?

**Review by makkes — CHANGES_REQUESTED** [review]

File: pkg/cmd/pkg/package_list.go (hunk 1 above)
Comment (makkes) [inline pkg/cmd/pkg/package_list.go]: After @bamarni chimed in I would prefer calling this column `CERTIFIED` which better explains its purpose.

**Review by makkes — CHANGES_REQUESTED** [review]

File: pkg/cmd/pkg/package_list.go (hunk 1 above)
Comment (janisz) [reply pkg/cmd/pkg/package_list.go]: This changes scope of this task

Comment (janisz) [reply pkg/cmd/pkg/package_list.go]: Fixed

**Review by makkes — APPROVED** [review]
awesome

File: pkg/cmd/pkg/package_list.go [hunk 2]
```diff
@@ -19,149 +21,175 @@ import (
...
+		}
+
+		description := p.Description
+		if strings.Index(description, "\n") > -1 {
+			i := strings.Index(description, "\n")
+			description = description[:i] + "..."
+		}
+		if utf8.RuneCountInString(description) > 70 {
+			description = description[0:66] + "..."
+		}
```
Comment (drewvanstone) [inline pkg/cmd/pkg/package_list.go:165]: I like this, very clean 👍

File: pkg/cmd/pkg/package_list.go [hunk 3]
```diff
@@ -19,149 +21,175 @@ import (
...
+			}
+			c, err := cosmos.NewClient(ctx, pluginutil.HTTPClient(""))
 			if err != nil {
 				return err
 			}
+			return listPackages(ctx, opts, c)
+		},
+	}
+	cmd.Flags().StringVar(&opts.appID, "app-id", "", "The application ID")
+	cmd.Flags().BoolVar(&opts.cliOnly, "cli", false, "Command line only")
```
Comment (drewvanstone) [inline pkg/cmd/pkg/package_list.go:63]: Hmm I think the description here should be a little clearer. What does "Command line only" mean? Does it disable some web interface or something?

File: pkg/cmd/pkg/package_list.go [hunk 4]
```diff
@@ -19,149 +21,175 @@ import (
...
+	cmd.Flags().BoolVar(&opts.jsonOutput, "json", false, "Print options json format")
```
Comment (drewvanstone) [inline pkg/cmd/pkg/package_list.go:64]: Wording here can be cleaned up a bit too. Maybe "Set output format to json"?

**Review by drewvanstone — CHANGES_REQUESTED** [review]
Just a couple small points, but overall this looks fantastic!

**Comment by janisz** [comment]
Flags descriptions should be consistent across whole project. How about fixing it in new PR?

```go
pkg/cmd/pkg/package_describe.go:        cmd.Flags().BoolVar(&opts.cliOnly, "cli", false, "Command line only")
pkg/cmd/pkg/package_install.go: cmd.Flags().BoolVar(&options.cliOnly, "cli", false, "Command line only")
pkg/cmd/pkg/package_list.go:    cmd.Flags().BoolVar(&opts.cliOnly, "cli", false, "Command line only")
pkg/cmd/pkg/package_uninstall.go:       cmd.Flags().BoolVar(&cliOnly, "cli", false, "Command line only")
```
```go
pkg/cmd/diagnostics/diagnostics_list.go:        cmd.Flags().BoolVar(&jsonOutput, "json", false, "Print in json format")
pkg/cmd/job/job_history.go:     cmd.Flags().BoolVar(&jsonOutput, "json", false, "Print in json format")
pkg/cmd/job/job_list.go:        cmd.Flags().BoolVar(&jsonOutput, "json", false, "Print in json format")
pkg/cmd/job/job_queue.go:       cmd.Flags().BoolVar(&jsonOutput, "json", false, "Print in json format")
pkg/cmd/job/job_schedule_show.go:       cmd.Flags().BoolVar(&jsonOutput, "json", false, "Print in json format")
pkg/cmd/job/job_show_runs.go:   cmd.Flags().BoolVar(&jsonOutput, "json", false, "Print in json format")
pkg/cmd/node/node_diagnostics_cancel.go:        cmd.Flags().BoolVar(&jsonOutput, "json", false, "Print in json format")
pkg/cmd/node/node_diagnostics_list.go:  cmd.Flags().BoolVar(&jsonOutput, "json", false, "Print in json format")
pkg/cmd/node/node_diagnostics_status.go:        cmd.Flags().BoolVar(&jsonOutput, "json", false, "Print in json format")
pkg/cmd/node/node_dns.go:       cmd.Flags().BoolVar(&jsonOutput, "json", false, "Print in json format")
pkg/cmd/node/node_list.go:      cmd.Flags().BoolVar(&jsonOutput, "json", false, "Print in json format")
pkg/cmd/node/node_list_components.go:   cmd.Flags().BoolVar(&jsonOutput, "json", false, "Print in json format")
pkg/cmd/node/node_metrics_details.go:   cmd.Flags().BoolVar(&jsonOutput, "json", false, "Print in json format")
pkg/cmd/node/node_metrics_summary.go:   cmd.Flags().BoolVar(&jsonOutput, "json", false, "Print in json format")
pkg/cmd/pkg/package_list.go:    cmd.Flags().BoolVar(&opts.jsonOutput, "json", false, "Print options json format")
pkg/cmd/pkg/package_repo_list.go:       cmd.Flags().BoolVar(&jsonOutput, "json", false, "Print in json format")
pkg/cmd/pkg/package_search.go:  cmd.Flags().BoolVar(&jsonOutput, "json", false, "Print in json format")
pkg/cmd/quota/quota_get.go:     cmd.Flags().BoolVar(&jsonOutput, "json", false, "Print in json format")
pkg/cmd/quota/quota_list.go:    cmd.Flags().BoolVar(&jsonOutput, "json", false, "Print in json format")
pkg/cmd/service/service_list.go:        cmd.Flags().BoolVar(&jsonOutput, "json", false, "Print in json format")
pkg/cmd/task/task_list.go:      cmd.Flags().BoolVar(&jsonOutput, "json", false, "Print in json format")
pkg/cmd/task/task_metrics_details.go:   cmd.Flags().BoolVar(&jsonOutput, "json", false, "Print in json format")
pkg/cmd/task/task_metrics_summary.go:   cmd.Flags().BoolVar(&jsonOutput, "json", false, "Print in json format")
```

**Review by drewvanstone — APPROVED** [review]

**Comment by makkes** [comment]
@mesosphere-ci run all integration tests

**Review by bamarni — APPROVED** [review]

**Comment by janisz** [comment]
@mesosphere-ci run linux integration tests

**Pull request merged**